### DISCUSSION TITLE
Security mechanisms in the winget repository. Using own repository.

### DISCUSSION THREAD

**Discussion by m-fessler** [microsoft/winget-cli#4039] [Q&A]
Hello everyone,

I have used winget very little so far as I am still waiting for the feature that the version of msstore apps can be determined and only these can be updated, but also I have (unjustified?) security concerns.

What security mechanisms exist for the winget repository?
What protects me from someone releasing a maliciously manipulated version of a software?
Be it an update, a similar entry or pretending to come from the original developer when they are not even represented in the repository.

Is there a way to use a repository that I create and maintain myself?

Thanks and greetings,
Martin

**Comment by mdanish-kh** [comment]
> What security mechanisms exist for the winget repository?

There is an [Open at Microsoft Episode (YouTube)](https://www.youtube.com/watch?v=b976Vdv2VAs&t=204s) (2:53 timestamp) where @denelon talks at length how an individual package is validated before it is merged into the community repository.

There are also related discussions that you can read:

- https://github.com/microsoft/winget-cli/discussions/4012
- https://github.com/microsoft/winget-pkgs/discussions/31547

> Is there a way to use a repository that I create and maintain myself?

There is a reference implementation available for a REST source for hosting winget packages
- https://github.com/microsoft/winget-cli-restsource/
